### ISSUE TITLE
before_send_check_in breaks serialization when using php artisan optimize

### ISSUE THREAD

**Issue by InterstellarStella** [getsentry/sentry-laravel#893]
### How do you use Sentry?

Sentry SaaS (sentry.io)

### SDK version

4.4.1

### Steps to reproduce

![image](https://github.com/getsentry/sentry-laravel/assets/81985895/45c90931-a835-43c4-aa9b-7c2c2c62bb51)
Then use `php artisan optimize`

### Expected result

The process is completed without errors.

### Actual result

We get the error `LogicException: Your configuration files are not serializable.`.
This error isn't thrown if we don't optimize.

Link to the error in the shadow ticket.

**Comment by stayallive** [comment]
See here on how to fix it: https://docs.sentry.io/platforms/php/guides/laravel/configuration/laravel-options/#closures-and-config-caching

**Comment by didac-adria** [comment]
Thank you guys. So I guess that by something like this should work:
<img width="1233" alt="image" src="https://github.com/getsentry/sentry-laravel/assets/63950600/d104355e-e0e0-41e5-bb0b-b52e41c74f54">

<img width="1362" alt="image" src="https://github.com/getsentry/sentry-laravel/assets/63950600/0bb6db67-9711-4d16-8e2d-9b298e68841a">

Have in mind that we need to call back to the sentry config file within the NinjaSentryServiceProvider because env variables must be called within config files only.

**Comment by stayallive** [comment]
Yes, however you can't add any custom config options to your `sentry.php` config file since our SDK will not accept that so you would need to keep those settings elsewhere in your `app.php` config file or something or `services.php`.

**Comment by didac-adria** [comment]
That's right. This is the problem I've just faced. I was about to report that. That's a pitty, we should keep things organized.
I've moved it to another config file and works now.
